### PR TITLE
Refactor code to move out fav/unfav from UpdateRun

### PR DESCRIPTION
**Pull request by jecepeda**
## Summary

* Add new GraphQL mutation to fav/unfav a run
* Remove IsFavorite from UpdateRun mutation
* Add tests in server to test the new functionality
* Update schema
* Change webapp code to use the new mutation

## Ticket Link

Closes https://github.com/mattermost/mattermost-server/issues/21706

## Checklist

- [ ] ~Telemetry updated~
- [ ] ~Gated by experimental feature flag~
- [x] Unit tests updated

### CODE REVIEW

File: webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx [hunk 1]
```diff
@@ -41,19 +41,19 @@ export const RunHeader = ({playbookRun, playbookRunMetadata, isFollowing, hasPer
     const dispatch = useDispatch();
     const {formatMessage} = useIntl();
     const updateRun = useUpdateRun(playbookRun.id);
-    const [isFavoriteRun, toggleFavorite] = useFavoriteRun(playbookRun.team_id, playbookRun.id);
+    const [isFavorite, toggleFavorite] = useFavoriteRun(playbookRun.team_id, playbookRun.id);
```
Comment (trilopin) [inline webapp/src/components/backstage/playbook_runs/playbook_run/header.tsx:44]: Nit: this rename is unnecessary, I would revert it to remove some noise.

**Review by trilopin — APPROVED** [review]
Great job moving this piece of code @jecepeda 🎉 

Now that fav/unfav is outside updateRun mutation, we could remove the refetch LHSdocument for that operation since it was the main reason (to have live updates in LHS categories). However, the run name is also affected. 

thoughts on keep/remove the refech @crspeller ?

**Comment by trilopin** [comment]
@jecepeda there are some conflicts because we recently merged a significant pr. Could you please update your branch with the current master and fix them?

no problem if you can't, please tell me and I'll happily do it.

**Comment by jecepeda** [comment]
Sure! Gonna do it 😊

Jaime Cepeda

On November 29, 2022, "Github: Notifications" ***@***.***>
wrote:
> @jecepeda <https://github.com/jecepeda> there are some conflicts
> because we recently merged a significant pr. Could you please update
> your branch with the current master and fix them?
>
> no problem if you can't, please tell me and I'll happily do it.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/mattermost/mattermost-plugin-
playbooks/pull/1615#issuecomment-1331140245>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/ABRQ2RGDBICSYRRFZHXHDJLWKZGRXANCNFSM6AAAAAASOEC5F4>.
You are receiving this because you were mentioned.Message ID:
<mattermost/mattermost-plugin-
***@***.***>

**Review by crspeller — APPROVED** [review]
Thanks for the contribution!

I think we can leave it in for now. It would be nice if the name updated.

**Comment by trilopin** [comment]
@jecepeda THank you again ❤️

**Pull request merged**